File: tests/test_v443_playback_endurance.js
'use strict';
const fs=require('fs'),path=require('path'),assert=require('assert');
const root=path.resolve(__dirname,'..');
const version=fs.readFileSync(path.join(root,'VERSION'),'utf8').trim();
const app=fs.readFileSync(path.join(root,'app.js'),'utf8');
const terminal=fs.readFileSync(path.join(root,'architecture/playback-terminal.js'),'utf8');
const index=fs.readFileSync(path.join(root,'index.html'),'utf8');
const css=fs.readFileSync(path.join(root,'ui/playback-terminal.css'),'utf8');
const core=fs.readFileSync(path.join(root,'core-model.js'),'utf8');

assert(app.includes('PLAYBACK_ENGINE_FAILURE_THRESHOLD=3'),'systemic startup failure threshold missing');
assert(app.includes('TRANSIENT_UNPLAYABLE_MEDIA=new Map()'),'transient failure quarantine missing');
assert(app.includes("window.SBB_PLAYBACK_ENGINE=Object.freeze"),'playback engine authority missing');
assert(app.includes('RECENT_HISTORY_AUTOFILL_DAYS=3'),'three-day recent-history refill missing');
assert(app.includes('currentGameKey:()=>programGameIdentity'),'endurance current-game hook missing');
assert(app.includes('currentIsFullRecap:()=>isFullRecapCandidate'),'endurance recap hook missing');

const completion=app.slice(app.indexOf('function advanceAfterCompletedItem'),app.indexOf('function advance(direction=1)'));
assert(completion.includes('!isFullRecapCandidate(finished)'),'completion must use semantic recap classification');
assert(!completion.includes('!finished?.overview'),'raw overview flag must not control recap completion');
const queue=app.slice(app.indexOf('function nextVisibleQueueIndex'),app.indexOf('function renderQueue'));
assert(queue.includes('!isFullRecapCandidate(current)'),'next queue item must use semantic recap classification');
assert(app.includes('preferGameOverviews(expandMediaVersions'),'score-card recap selection must compare alternate versions');

assert(terminal.includes('FIRST_FRAME_WATCHDOG_MS=28_000'),'first-frame watchdog missing');
assert(terminal.includes('DUPLICATE_GAME_RECAP'),'same-game alternate recap failure guard missing');
assert(terminal.includes('UNRECOVERABLE_NO_FIRST_FRAME'),'unrecoverable no-frame failure guard missing');
assert(terminal.includes('chaosDisruptStandby'),'hammer standby disruption missing');
assert(terminal.includes('forcePlaybackEngineReset'),'automatic playback engine reset missing');
assert(index.includes('id="playbackEnduranceStart"'),'terminal endurance start control missing');
assert(index.includes('id="playbackEnduranceStop"'),'terminal endurance stop control missing');
assert(index.includes(`architecture/playback-terminal.js?v=${version}`),'terminal cache-bust version stale');
assert(index.includes(`app.js?v=${version}`),'app cache-bust version stale');
assert(css.includes('.pt-endurance'),'endurance terminal styling missing');
assert(core.includes(`version:'${version}'`),'core model version stale');
console.log(`PASS: v${version} retains the v4.4.3 playback-resilience baseline`);

File: tests/test_v443_playback_endurance.py
import unittest
from pathlib import Path
ROOT=Path(__file__).resolve().parents[1]
APP=(ROOT/'app.js').read_text(encoding='utf-8')
TERM=(ROOT/'architecture'/'playback-terminal.js').read_text(encoding='utf-8')
INDEX=(ROOT/'index.html').read_text(encoding='utf-8')
CORE=(ROOT/'core-model.js').read_text(encoding='utf-8')

class V443PlaybackEndurance(unittest.TestCase):
    def test_semantic_recap_completion_prevents_same_game_alternate(self):
        block=APP[APP.index('function advanceAfterCompletedItem'):APP.index('function advance(direction=1)')]
        self.assertIn('!isFullRecapCandidate(finished)',block)
        self.assertNotIn('!finished?.overview',block)
    def test_transient_failures_do_not_permanently_poison_assets(self):
        block=APP[APP.index('function markRuntimeMediaFailed'):APP.index('function mediaMatchesScoreGame')]
        self.assertIn('playbackFailureIsAssetSpecific(reason)',block)
        self.assertIn('noteTransientPlaybackFailure(item,reason)',block)
        self.assertIn('RUNTIME_UNPLAYABLE_MEDIA.add(key)',block)
    def test_engine_circuit_breaker_and_zero_extra_decoders(self):
        self.assertIn('PLAYBACK_ENGINE_FAILURE_THRESHOLD=3',APP)
        self.assertIn('function resetPlaybackEngine',APP)
        prepared=APP[APP.index('function scorePreparedLimit'):APP.index('function scoreServerWarmLimit')]
        self.assertIn('return 0;',prepared)
    def test_recent_final_refill_is_three_days(self):
        self.assertIn('RECENT_HISTORY_AUTOFILL_DAYS=3',APP)
        self.assertIn('scheduleRecentHistoricalRecapFill(date)',APP)
    def test_terminal_endurance_phases_and_failure_guards(self):
        self.assertIn("label:'WARMUP',durationMs:5*60_000",TERM)
        self.assertIn("label:'SOAK',durationMs:15*60_000",TERM)
        self.assertIn("label:'HAMMER',durationMs:10*60_000",TERM)
        self.assertIn('DUPLICATE_GAME_RECAP',TERM)
        self.assertIn('UNRECOVERABLE_NO_FIRST_FRAME',TERM)
        self.assertIn('chaosDisruptStandby',TERM)
    def test_release_version_and_controls(self):
        self.assertEqual((ROOT/'VERSION').read_text().strip(),'4.4.3')
        self.assertIn('playbackEnduranceStart',INDEX)
        self.assertIn('playbackEnduranceStop',INDEX)
        self.assertIn('app.js?v=4.4.3',INDEX)
        self.assertIn("version:'4.4.3'",CORE)

if __name__=='__main__': unittest.main()
